Sources/DistributedActors/Time.swift

```diff
@@ -18,116 +18,72 @@ import struct Foundation.Date
-    ///
-    /// - parameters:
-    ///     - amount: the amount of milliseconds this `TimeAmount` represents.
-    /// - returns: the `TimeAmount` for the given amount.
-    public static func milliseconds(_ amount: Value) -> TimeAmount {
-        TimeAmount(amount * 1000 * 1000)
+    /// The microseconds representation of the `TimeAmount`.
+    public var microseconds: Int64 {
+        self.nanoseconds / TimeUnit.microseconds.rawValue
     }
 
-    public static func milliseconds(_ amount: Int) -> TimeAmount {
-        self.milliseconds(Value(amount))
+    /// The milliseconds representation of the `TimeAmount`.
+    public var milliseconds: Int64 {
+        self.nanoseconds / TimeUnit.milliseconds.rawValue
     }
 
-    /// Creates a new `TimeAmount` for the given amount of seconds.
-    ///
-    /// - parameters:
-    ///     - amount: the amount of seconds this `TimeAmount` represents.
-    /// - returns: the `TimeAmount` for the given amount.
-    public static func seconds(_ amount: Value) -> TimeAmount {
-        TimeAmount(amount * 1000 * 1000 * 1000)
+    /// The seconds representation of the `TimeAmount`.
+    public var seconds: Int64 {
+        self.nanoseconds / TimeUnit.seconds.rawValue
     }
 
-    public static func seconds(_ amount: Int) -> TimeAmount {
-        self.seconds(Value(amount))
+    public var isEffectivelyInfinite: Bool {
+        self.nanoseconds == .max
     }
+}
 
-    /// Creates a new `TimeAmount` for the given amount of minutes.
+extension Duration {
+    /// Creates a new `Duration` for the given amount of minutes.
     ///
     /// - parameters:
-    ///     - amount: the amount of minutes this `TimeAmount` represents.
-    /// - returns: the `TimeAmount` for the given amount.
-    public static func minutes(_ amount: Value) -> TimeAmount {
-        TimeAmount(amount * 1000 * 1000 * 1000 * 60)
+    ///     - amount: the amount of minutes this `Duration` represents.
+    /// - returns: the `Duration` for the given amount.
+    public static func minutes(_ amount: Value) -> Duration {
+        .nanoseconds(amount * 1000 * 1000 * 1000 * 60)
     }
 
-    public static func minutes(_ amount: Int) -> TimeAmount {
+    public static func minutes(_ amount: Int) -> Duration {
         self.minutes(Value(amount))
     }
 
-    /// Creates a new `TimeAmount` for the given amount of hours.
+    /// Creates a new `Duration` for the given amount of hours.
     ///
     /// - parameters:
-    ///     - amount: the amount of hours this `TimeAmount` represents.
-    /// - returns: the `TimeAmount` for the given amount.
-    public static func hours(_ amount: Value) -> TimeAmount {
-        TimeAmount(amount * 1000 * 1000 * 1000 * 60 * 60)
+    ///     - amount: the amount of hours this `Duration` represents.
+    /// - returns: the `Duration` for the given amount.
+    public static func hours(_ amount: Value) -> Duration {
+        .nanoseconds(amount * 1000 * 1000 * 1000 * 60 * 60)
     }
```

Comment:
This always felt a bit silly to be dropped off from the official type... thanks for adding them

I wonder if we "have to" make them internal...

Reply:
(all the .hours / .seconds thingies i mean)

Reply:
Made these internal